packages/eyeglass/src/assets/AssetsSource.ts

```diff
@@ -22,69 +29,85 @@ var stringify = require("json-stable-stringify");
  * Option: globOpts [Optional] - Options to use for globbing.
  *   See: https://github.com/isaacs/node-glob#options
  */
-export default function AssetsSource(srcPath, options) {
-  options = options || {};
+class AssetsSource {
+  name: string | null;
+  httpPrefix: string | null;
```

Comment:
They are optional but the code explicitly casts `undefined` to `null` here: https://github.com/sass-eyeglass/eyeglass/blob/type_annotations/packages/eyeglass/src/assets/AssetsSource.ts#L45-L47

Because this PR is mostly about documenting the existing code, I didn't try to refactor things like this.